BuddySave/System/ProcessProvider.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Diagnostics;
+
+namespace BuddySave.System;
+
+internal class ProcessProvider : IProcessProvider
+{
+    public Process Start(ProcessStartInfo processStartInfo)
+    {
+        return Process.Start(processStartInfo);
+    }
+
+    public async Task WaitForExitAsync(Process process)
+    {
+        await process.WaitForExitAsync();
```

Comment:
Shouldn't we pass down the cancelation token?

Reply:
Passing cancellation token would allow stop waiting for server stop. Essentially disabling auto-save. Additionally right now buddy save waits on the main thread, so no console input is read (unless ctrl+c would work).

So not sure it's worth using cancellation token

Reply:
It's not that well explanatory how all of this ties together. Then maybe it's worth to describe it with a summary comment in GameSession so the intention would be clear and we don't forget how it was intended to work? :)